### PR TITLE
DAOS-12070 il: comply with fgetws() API

### PR DESCRIPTION
**Pull request by bfaccini**
for some reason the dfuse_fgetws() version was missing the 2nd parameter, causing crashes when it was finally dereferenced as a FILE stream when it was an int/length in fact.

Required-githooks: true

Signed-off-by: Bruno Faccini <bruno.faccini@intel.com>

### Before requesting gatekeeper:

* [ ] Two review approvals and any prior change requests have been resolved.
* [ ] Testing is complete and all tests passed or there is a reason documented in the PR why it should be force landed and forced-landing tag is set.
* [ ] `Features:` (or `Test-tag*`) commit pragma was used or there is a reason documented that there are no appropriate tags for this PR.
* [ ] Commit messages follows the guidelines outlined [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments).
* [ ] Any tests skipped by the ticket being addressed have been run and passed in the PR.

### Gatekeeper:

* [ ] You are the appropriate gatekeeper to be landing the patch.
* [ ] The PR has 2 reviews by people familiar with the code, including appropriate watchers.
* [ ] Githooks were used. If not, request that user install them and check copyright dates.
* [ ] Checkpatch issues are resolved.  Pay particular attention to ones that will show up on future PRs.
* [ ] All builds have passed.  Check non-required builds for any new compiler warnings.
* [ ] Sufficient testing is done. Check feature pragmas and test tags and that tests skipped for the ticket are run and now pass with the changes.
* [ ] If applicable, the PR has addressed any potential version compatibility issues.
* [ ] Check the target branch.   If it is master branch, should the PR go to a feature branch?  If it is a release branch, does it have merge approval in the JIRA ticket.
* [ ] Extra checks if forced landing is requested
  * [ ] Review comments are sufficiently resolved, particularly by prior reviewers that requested changes.
  * [ ] No new NLT or valgrind warnings.  Check the classic view.
  * [ ] Quick-build or Quick-functional is not used.
* [ ] Fix the commit message upon landing. Check the standard [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments). Edit it to create a single commit. If necessary, ask submitter for a new summary.

### CODE REVIEW

File: src/client/dfuse/il/int_posix.c [hunk 1]
```diff
@@ -2493,7 +2493,7 @@ dfuse_fgets(char *str, int n, FILE *stream)
 }
 
 DFUSE_PUBLIC wchar_t *
-dfuse_fgetws(const wchar_t *ws, FILE *stream)
+dfuse_fgetws(const wchar_t *ws, int n, FILE *stream)
```
Comment (ashleypittman) [inline src/client/dfuse/il/int_posix.c:2496]: Can you remove the D_ERROR call on L2502 as well please.

Comment (bfaccini) [reply src/client/dfuse/il/int_posix.c:2496]: Done in next commit

File: src/client/dfuse/il/intercept.h [hunk 2]
```diff
@@ -8,6 +8,7 @@
 #define __INTERCEPT_H__
 #include <unistd.h>
 #include <stdlib.h>
+#include <wchar.h>
```
Comment (ashleypittman) [inline src/client/dfuse/il/intercept.h:11]: Does this make a difference?  It seems like it's probably a good idea though.

**Review by ashleypittman — COMMENTED** [review]
Can you re-push this with "Features: dfuse" to verify that it passes the dfuse/il tests please.

**Review by jolivier23 — DISMISSED** [review]

**Comment by bfaccini** [comment]
Only test failure of last CI session is : 
```
Existing failures - 1
Test Hardware / Functional Hardware Medium / 1-./mdtest/small.py:MdtestSmall.test_mdtest_small;run-agent_config-transport_config-container-dfuse-dmg-hosts-mdtest-client_processes-pool-server_config-engines-0-storage-0-1-7dc5 – FTEST_mdtest.MdtestSmall
```
which looks like issue already being tracked by [DAOS-11203](https://daosio.atlassian.net/browse/DAOS-11203) where, for some unknown reason, Mdtest just runs slow and triggers timeout.

**Comment by bfaccini** [comment]
> Can you re-push this with "Features: dfuse" to verify that it passes the dfuse/il tests please.

@ashleypittman can you check by yourself in last commit's CI tests results if this has occurred as you expected, because I am fed up searching in these CI artefacts where the files names/organisation is just a mess

**Comment by ashleypittman** [comment]
> > Can you re-push this with "Features: dfuse" to verify that it passes the dfuse/il tests please.
> 
> @ashleypittman can you check by yourself in last commit's CI tests results if this has occurred as you expected, because I am fed up searching in these CI artefacts where the files names/organisation is just a mess

https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-10877/8/testReport/FTEST_dfuse/DaosBuild/Test___Functional_on_EL_8___1___dfuse_daos_build_py_DaosBuild_test_dfuse_daos_build_wt_il_run_container_dfuse_dfuse_vm_hosts_pool_server_config_engines_0_storage_0_b457/ is the one I was looking for.

I think we can land this PR now.

**Comment by bfaccini** [comment]
> I think we can land this PR now.

Thanks for your help @ashleypittman !

**Review by ashleypittman — DISMISSED** [review]

**Review by ashleypittman — APPROVED** [review]

**Comment by jolivier23** [comment]
can you merge latest master?   Looks like the Ubuntu build failed so no tests ran.  This is a pretty old problem we were hitting.

**Review by Michael-Hennecke — APPROVED** [review]

**Comment by bfaccini** [comment]
> can you merge latest master? Looks like the Ubuntu build failed so no tests ran. This is a pretty old problem we were hitting.

done with latest commit

**Comment by bfaccini** [comment]
CI run #10 had the single test failure following : 
```
New failing - 1
Test Hardware / Functional Hardware Medium UCX Provider / IO31: fetch when all replicas unavailable – FTEST_daos_test.DAOS_IO_EC_4P2G1
Stacktrace
0 != 0xfffffffffffffc13
src/tests/suite/daos_obj.c:476: error: Failure!
```
but looking into all the logs saved in the artifacts I do not find any evidence of an error :-(
And when I have restarted the "Test Hardware" stage, only the previously failed "Functional Hardware Medium UCX Provider" step has been reran, and it has been successful this time !!...

**Comment by johannlombardi** [comment]
is that ready to land or does the failure reported above needs further investigation?

**Comment by bfaccini** [comment]
> is that ready to land or does the failure reported above needs further investigation?

Well it should be ready to land, only thing I am not sure is that if @Michael-Hennecke had finally the opportunity to test this change against the original problem reproducer at Lenovo ...

**Comment by ashleypittman** [comment]
I think we should land this.

**Pull request merged**